src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -388,52 +388,92 @@ public override void RegisterNativeMembers (JniType nativeClass, Type type, stri
 					return;
 				}
 
-				string[] members = methods!.Split ('\n');
-				if (members.Length < 2) {
+				int newLineIndex = methods!.IndexOf ('\n');
+				if (newLineIndex == -1) {
 					if (jniAddNativeMethodRegistrationAttributePresent)
 						base.RegisterNativeMembers (nativeClass, type, methods);
 					return;
 				}
 
-				JniNativeMethodRegistration[] natives = new JniNativeMethodRegistration [members.Length-1];
-				for (int i = 0; i < members.Length; ++i) {
-					string method = members [i];
-					if (string.IsNullOrEmpty (method))
-						continue;
-					string[] toks = members [i].Split (new[]{':'}, 4);
-					Delegate callback;
-					if (toks [2] == "__export__") {
-						var mname = toks [0].Substring (2);
-						MethodInfo? minfo = null;
-						foreach (var mi in type.GetMethods (BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static))
-							if (mi.Name == mname && JavaNativeTypeManager.GetJniSignature (mi) == toks [1]) {
-								minfo = mi;
-								break;
+				List<JniNativeMethodRegistration> natives = new List<JniNativeMethodRegistration> (16); // an initial capacity since most classes have more than 4-8 methods
+				ReadOnlySpan<char> methodsSpan = methods;
+
+				MethodInfo []? typeMethods = null;
+				while (!methodsSpan.IsEmpty) {
+					ReadOnlySpan<char> methodLine = methodsSpan.Slice (0, newLineIndex != -1 ? newLineIndex : methodsSpan.Length);
+					if (!methodLine.IsEmpty) {
+						SplitMethodLine (methodLine,
+							out ReadOnlySpan<char> name,
+							out ReadOnlySpan<char> signature,
+							out ReadOnlySpan<char> callbackString,
+							out ReadOnlySpan<char> callbackDeclaringTypeString);
+						Delegate callback;
+						if (callbackString.SequenceEqual ("__export__")) {
+							var mname = name.Slice (2);
+							MethodInfo? minfo = null;
+							typeMethods ??= type.GetMethods (BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static);
```

Comment:
This shouldn't need to look at Public or Instance methods.

Reply:
These are the existing binding flags being used. I'm not inclined to changing that with this PR. I believe something like that should be a separate change.

The only thing that I changed here is to cache the call to `type.GetMethods` into a local variable, since the `type` never changes. Calling `type.GetMethods` each time in the loop doesn't make sense.